geode-core/src/main/java/org/apache/geode/distributed/internal/DistributionStats.java

```diff
@@ -1600,17 +1594,14 @@ public long startReplyWait() {
     return getStatTime();
   }
 
+
   @Override
   public void endReplyWait(long startNanos, long initTime) {
     if (enableClockStats) {
       stats.incLong(replyWaitTimeId, getStatTime() - startNanos);
       // this.replyWaitHistogram.endOp(delta);
-    }
-    if (initTime != 0) {
-      long mswait = System.currentTimeMillis() - initTime;
-      if (mswait > getReplyWaitMaxTime()) {
-        stats.setLong(replyWaitMaxTimeId, mswait);
-      }
+      long currentWaitTime = System.currentTimeMillis() - initTime;
```

Comment:
The maxReplyWaitTime should not be in "enableClockStats". This stat is intentionally still enabled even when time stats are disabled. That is why it uses System.currentTimeMillis() instead of getStatTime().
I think we should preserve this behaviour for this pull request.

Reply:
okay, that makes sense.  the original `if (initTime != 0) {` check probably isn't needed, right?

Reply:
I think we still need it or we need to refactor DistributionStats.endReplyWait into two methods.
If you look at the call of endReplyWait in: org.apache.geode.distributed.internal.direct.DirectChannel.sendToMany(MembershipManager, InternalDistributedMember[], DistributionMessage, long, long)
you will see that it passes in an initTime of 0 if ackTimeout is <= 0.
So this caller is actually using an initTime of 0 to see don't bother updating the maxReplyWaitTime. I would vote for splitting endReplyWait into two methods but this will cause you to change the two classes that call endReplyWait.

Reply:
i don't want to do that refactor on this commit. fair?

in order to change as little as possible, i left the `if(initTime!=0)` check

Reply:
sounds good